modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -521,6 +538,76 @@ static Response innerShardOperation(Request request, ScriptService scriptService
                     double result = scoreScript.execute(null);
                     return new Response(result);
                 }, indexService);
+            } else if (scriptContext == BooleanFieldScript.CONTEXT) {
+                return prepareRamIndex(request, (context, leafReaderContext) -> {
+                    BooleanFieldScript.Factory factory = scriptService.compile(request.script, BooleanFieldScript.CONTEXT);
+                    BooleanFieldScript.LeafFactory leafFactory =
+                            factory.newFactory("boolean_runtime_field", request.getScript().getParams(), context.lookup());
+                    BooleanFieldScript booleanFieldScript = leafFactory.newInstance(leafReaderContext);
+                    booleanFieldScript.setDocument(0);
+                    booleanFieldScript.execute();
+                    return new Response(Map.of("trues", booleanFieldScript.trues(), "falses", booleanFieldScript.falses()));
```

Comment:
This seems odd but I realize it is a result of the existing output of BooleanFieldScript. @nik9000 What about using a list of values like the other runtime field types do?

Reply:
I talked to @jdconrad about this yesterday and I don't have a great answer here. I like that `BooleanFieldScript` can do what it wants, at least on the theoretical level. And I don't think it's worth modifying `BooleanFieldScript` just for this API. The ordering of the emitted booleans isn't really important in any other context.

But I think it'd be entirely reasonable for this API to make a list out of the counts just to make it consistent. That is also consistent with the way doc values and the `fields` API work for this.

I do think its nice if we give these scripts the freedom to change their minds about they accumulate values without it breaking folks. I can't think of how they might do that right now but that don't mean there aren't good ways. If you "work like doc values" in this API then we have a consistent way to send the results to the callers.

As it stands now the scripts preserve ordering of values for everything but booleans. But the rest of ES leaves us the option to change that ordering in the future. And I'd like to keep it.

Reply:
I just stumbled upon the same as part of adding support for script to BooleanFieldMapper and I went for doing what doc_values do, meaning returning false values first, then true values.Given that doc_values are resorted, and the fields API does not guarantee ordering, it looks like it is not needed to hold an array of values for this specific case. What we could do is expose a method that returns the values in whatever order we decide to return them, so that consumers don't have to do it themselves. And I would not expose in the response the trues vs falses concept, but rather expose an array of values, regardless of whether we have it internally or not.

Reply:
Like I said, I don't think we should be exposing trues and falses here, but an array of values like we do for other types. I am adding a method to achieve the same [here](https://github.com/elastic/elasticsearch/pull/71454/files#diff-0938096fde41d31361acc4b51c62a1012aa99e97732d8f7159a1ca8fd9664aebR49), can we do something similar?